utils/build.mjs

```diff
@@ -59,7 +59,7 @@ export async function tsc() {
   const rootNodeModulesBin = fileURLToPath(
     new URL('../node_modules/.bin', import.meta.url)
   );
-  await execa('tsc', {
+  await execa('tsc', ['--declaration', '--emitDeclarationOnly'], {
```

Comment:
No because the `type-check` scripts all run `tsc --noEmit` which cannot be combined with `emitDeclarationOnly`. This means we need to explicitly pass in these flags given the context.

Reply:
Ok make sense. And it seems like `--declaration` is here because we want it for builders but we don't want it for cli 👍